shapely/tests/test_geometry.py

```diff
@@ -200,8 +206,29 @@ def test_get_z_2d():
 
 @pytest.mark.parametrize("geom", all_types)
 def test_new_from_wkt(geom):
+    assert_geometries_not_equal = False
+    if geom.is_empty and shapely.get_num_geometries(geom) > 0:
```

Comment:
- This logic is repeated a few times. Maybe it is good to have a centralized function that checks if the empty-collection / GEOS version conditions are met?
- The `xfail` means that the test is executed, but expected to fail. Isn't the `skip` better in this case?

Reply:
- Logic for checking equal geometries that yield unequal result is generalized into a function
- Some of the `pytest.xfail` conditions are changed to `pytest.skip` (perhaps more should?)